### DISCUSSION TITLE
Confused about css

### DISCUSSION THREAD

**Discussion by SimunKaracic** [com-lihaoyi/scalatags#272] [Q&A] [answered]
Hello!

I'm currently using scalatags to go through the [hypermedia systems](https://hypermedia.systems/) book, and it's been going great. I was able to easily figure out how to create custom htmx attributes, and it's generally just been a joy to use.

This, however, has me stumped. I have to add the following style to the document.

```
tr.htmx-swapping {
  opacity: 0;
  transition: opacity 1s ease-out;
}
```

Any ideas? I feel like I'm missing something super obvious

**Comment by lolgab** [comment, marked as answer]
I would just write a plain css file, serve it with the server and add it to the page with `link(rel := "stylesheet", href := "/main.css")`

**Comment by SimunKaracic** [comment]
Thanks, that works like a charm :) 
I was just hoping there's a way to keep it all in scalatags, but this is cool too

**Comment by PerWiklander** [comment]
There's also [ScalaCSS](https://github.com/japgolly/scalacss?tab=readme-ov-file) if you want a Scala DSL for type safe CSS.
See: https://japgolly.github.io/scalacss/book/quickstart/standalone.html

**Comment by PerWiklander** [comment]
BTW, it would be nice to trade experiences with Scalatags and htmx later on. Do you have any code online?
